ui: statement details import

koorosh opened the pull request:
Related to https://github.com/cockroachdb/cockroach/issues/57372
Depends on https://github.com/cockroachdb/ui/pull/162
Depends on https://github.com/cockroachdb/yarn-vendored/pull/51

Statement details and Diagnostics view pages are moved out
to `admin-ui-components` package and now they are imported back
as a reusable components.

In addition to changed imports, StatementDetails component now exposes
additional props for Diagnostics view tab. Before, StatementDetails
component used Diagnostics view connected to store and now it uses
pure component. This change were made to provide single point of
integration to redux store (the same way as other components connected).

Tracking analytics functionality for Diagnostics view was initially called directly
from components and now it's refactored out of component and involved
with redux actions and sagas. It was necessary to exclude this logic from pure
components and allow client apps which integrate Statement details to implement
their own logic for tracking analytics if necessary.

TODO:
- [x]  remove all exported components from this code base to avoid duplicates.
- [x]  update `admin-ui-components` package version in `package.json`

Release note: None

koorosh committed:
ui: re-import statement details page

Statement details and Diagnostics view pages are moved out
to `admin-ui-components` package and now they are imported back
as a reusable components.

In addition to changed imports, StatementDetails component now exposes
additional props for Diagnostics view tab. Before, StatementDetails
component used Diagnostics view connected to store and now it uses
pure component. This change were made to provide single point of
integration to redux store (the same way as other components connected).

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/58701)
<!-- Reviewable:end -->

dhartunian reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/> 

> Reviewed 8 of 8 files at r1, 10 of 10 files at r2.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/58701#-:-MQrV0Zb415mr-O5dSSm:bg7igkv)* status: :shipit: complete! 1 of 0 LGTMs obtained



<!-- Sent from Reviewable.io -->

koorosh committed:
ui: reuse components from admin-ui-components package

Currently, many components were exported to `admin-ui-components`
package and these components weren't removed from this repo.
To avoid duplicates, these components removed and re-imported back.

Release note: None

koorosh commented:
Lint errors can be fixed with this change https://github.com/cockroachdb/ui/pull/172

koorosh commented:
bors r+
